src/app/network/UploadManager.ts

```diff
@@ -43,11 +46,12 @@ type UploadManagerFileParams = {
 
 export const uploadFileWithManager = (
   files: UploadManagerFileParams[],
+  dispatch: Dispatch,
```

Comment:
We are coupling the upload manager seriously with other unrelated things. It would be better to add callbacks / event emitters to specific events, there is no need to generate more debt here.

Reply:
You are right. I didn't think at the time to put a callback, I change it.